CalibFormats/HcalObjects/src/HcalCalibrationWidthsSet.cc

```diff
@@ -13,11 +13,13 @@ const HcalCalibrationWidths& HcalCalibrationWidthsSet::getCalibrationWidths(cons
   std::vector<Item>::const_iterator cell;
   if (sorted_) {
     cell = std::lower_bound (mItems.begin(), mItems.end(), target);
-  }
-  else {
+  } else {
     cell = std::find(mItems.begin(),mItems.end(), target);
   }
-  if (cell == mItems.end() || cell->id != fId) 
+  if (cell == mItems.end() || 
+      ((fId.det()==DetId::Hcal && HcalDetId(cell->id) != HcalDetId(fId)) ||
+       (fId.det()==DetId::Calo && fId.subdetId()==HcalZDCDetId::SubdetectorId && HcalZDCDetId(cell->id) != HcalZDCDetId(fId)) ||
+       (fId.det()!=DetId::Hcal && (fId.det()==DetId::Calo && fId.subdetId()!=HcalZDCDetId::SubdetectorId) && (cell->id != fId))))
```

Comment:
@bsunanda this check is duplicated other times [here](https://github.com/cms-sw/cmssw/pull/8998/files#diff-9b72491dcbc3291a451ae85e443ad1a8R19) and [here](https://github.com/cms-sw/cmssw/pull/8998/files#diff-c23ed37c1513eac6c543d8ce62d84960R163) and [here](https://github.com/cms-sw/cmssw/pull/8998/files#diff-c23ed37c1513eac6c543d8ce62d84960R180).
In order to avoid code duplication, can this become a utility function in the Condition package?